service/ws/server_test.go

```diff
@@ -477,3 +477,33 @@ func TestSendMessages(t *testing.T) {
 
 	wg.Wait()
 }
+
+func TestRaceSendClose(t *testing.T) {
+	s, _, shutdown := setupServer(t)
+
+	var wg sync.WaitGroup
+	wg.Add(2)
+
+	go func() {
+		defer wg.Done()
+		for i := 0; i < 100; i++ {
+			_ = s.Send(Message{
+				ConnID: "connID",
+				Data:   []byte("some data"),
+				Type:   TextMessage,
+			})
+		}
+	}()
+
+	go func() {
+		defer wg.Done()
+		for i := 0; i < 100; i++ {
+			s.Close()
+		}
+		shutdown()
```

Comment:
Just for my own interest: I can't figure out what this is testing, exactly... :)

Reply:
@cpoile We are simply testing concurrent calls to `server.Close()` and `server.Send()` and iterating a while to make it more likely to trigger a race.

Reply:
I guess I'm wondering why the 100 `s.Close()`s? After the first one fires, the remaining `s.Send`s will silently error, right? (again, sorry, just trying to understand :) )